src/provider.tsx

```diff
@@ -92,14 +94,14 @@ export const IntercomProvider = ({
   }, []);
 
   const shutdown = React.useCallback(() => {
-    if (!isBooted.current) return;
+    if (isSSR || !isBooted.current) return;
```

Comment:
What is the specific reason for this check? It seems there is no use case such as:
"Intercom would be booted in a web environment and then suddenly wants to shutdown in a "not" web environment"?

Reply:
If SSR environment is not checked at this point, the IntercomAPI method is invoked, which will error on the use of `window`.

An alternative is to move the SSR check to the IntercomAPI method here, to save a few lines of code:
https://github.com/devrnt/react-use-intercom/blob/master/src/api.ts#L12

Reply:
Still think this can be dropped, because we will never get passed through `isBooted.current` (so the IntercomAPI method will never be invoked). 
We have to boot before shutting down and because `isSSR` is already checked in the `boot` method this seems unnecessary. 

In general I don't see a use case where SSR environment would not be checked at this point, right?

Reply:
The initial state of `isBooted.current` is set to the value of the `autoboot` option, so there is a scenario where it can be `true`.

To keep the logic simple, I think it just makes sense to check for `isSSR` where there's a reference to window, so I'd advocate removing the checks in the callbacks and just putting one check in here: 
https://github.com/devrnt/react-use-intercom/blob/master/src/api.ts#L12

Your call!

Reply:
Yes indeed. I would suggest to move the logic to `IntercomAPI`, thanks!